src/serverenvironment.cpp

```diff
@@ -1278,34 +1312,46 @@ void ServerEnvironment::step(float dtime)
 	/*
 		Step active objects
 	*/
-	{
+	if (m_active_object_interval_overload_skip > 0) {
+		ScopeProfiler sp(g_profiler, "SEnv: ActiveObject overload skips");
+		m_active_object_interval_overload_skip--;
+		// make sure we still accumulate the time for correct movements
+		m_accdtime += dtime;
+	} else {
+		m_accdtime += dtime;
 		ScopeProfiler sp(g_profiler, "SEnv: step act. objs avg", SPT_AVG);
-		//TimeTaker timer("Step active objects");
+		TimeTaker timer("Step active objects");
 
 		g_profiler->avg("SEnv: num of objects", m_active_objects.size());
 
 		// This helps the objects to send data at the same time
 		bool send_recommended = false;
-		m_send_recommended_timer += dtime;
+		m_send_recommended_timer += m_accdtime;
 		if(m_send_recommended_timer > getSendRecommendedInterval())
 		{
 			m_send_recommended_timer -= getSendRecommendedInterval();
 			send_recommended = true;
 		}
-
 		for (auto &ao_it : m_active_objects) {
 			ServerActiveObject* obj = ao_it.second;
 			if (obj->isGone())
 				continue;
 
 			// Step object
-			obj->step(dtime, send_recommended);
+			obj->step(m_accdtime, send_recommended);
 			// Read messages from object
 			while (!obj->m_messages_out.empty()) {
 				m_active_object_messages.push(obj->m_messages_out.front());
 				obj->m_messages_out.pop();
 			}
 		}
+		u32 time_ms = timer.stop(true);
+		u32 max_time_ms = 5;
```

Comment:
5ms tested well on my machine.
I tried with the nssm mod, which usually stops generating blocks with too many mobs. It remained playable.